File: competitive_programming/leetcode/06_javascript/1480_running_sum.js
function runningSum(nums) {
  const result = [];
  let total = 0;
  for (const value of nums) {
    total += value;
    result.push(total);
  }
  return result;
}

if (require.main === module) {
  if (runningSum([1, 2, 3, 4]).join(",") !== "1,3,6,10") throw new Error("running sum failed");
  if (runningSum([1, 1, 1, 1, 1]).join(",") !== "1,2,3,4,5") throw new Error("running sum failed");
  console.log("1480 running sum ok");
}

File: competitive_programming/leetcode/11_golang/1480_running_sum.go
package main

import "fmt"

func runningSum(nums []int) []int {
	result := make([]int, len(nums))
	total := 0
	for i, value := range nums {
		total += value
		result[i] = total
	}
	return result
}

func main() {
	if fmt.Sprint(runningSum([]int{1, 2, 3, 4})) != "[1 3 6 10]" || fmt.Sprint(runningSum([]int{1, 1, 1, 1, 1})) != "[1 2 3 4 5]" {
		panic("running sum failed")
	}
	fmt.Println("1480 running sum ok")
}
